docs/get-schedule-unsubscriptions.md

```diff
@@ -0,0 +1,84 @@
+# listPendingUnsubscriptions
+
+This example illustrates how we can leverage `polkadotjs` SDK to retrieve the list of all scheduled unsubscriptions
+into the worker node decentralized system.
+
+```js
+import { hexToU8a } from "@polkadot/util";
+import { ApiPromise, WsProvider } from "@polkadot/api";
+
+const wsProvider = new WsProvider(
+	"wss://ewx-dev-parachain-aule-qb9wx41jvm.energyweb.org/ws"
+);
+
+async function listPendingUnsubscriptions(): Promise<void> {
+	const api = await ApiPromise.create({ provider: wsProvider });
+
+	// Retrieve all scheduled tasks
+	const scheduledTasks = await api.query.scheduler.agenda.entries();
+
+	// Filter out pending unsubscriptions
+	const pendingScheduled = scheduledTasks.flatMap(([blockNumber, agenda]) => {
```

Comment:
So this is listing the scheduled unsubscribes (still usefull). Maybe i miss communicated on this.
We are mostly interested to be able to calculate the schedule name of the unsubscribe (when knowing the operator address and the solution group) and then get the block number that this unsubscribe will take effect.

Reply:
So this is the script to calculate the schedule name:
```
onst { blake2AsHex } = require('@polkadot/util-crypto');
const { u8aConcat, u8aToHex, stringToU8a } = require('@polkadot/util');
const { TypeRegistry, Text } = require('@polkadot/types');

const operator = '5EnT9tFabL7hoAsXTB5D7cmU9x944LpzT2QUUCY2gZaN5hxH'; // Replace with actual value
const solutionGroupNamespace = 'fokk3rm15'; // Replace with actual value

const registry = new TypeRegistry();
const operatorEncoded = registry.createType('AccountId', operator);
const namespaceEncoded = registry.createType(
  'Vec<u8>',
  u8aToHex(stringToU8a(solutionGroupNamespace)),
);

const unsubscribeEncoded = new Text(registry, 'Unsubscribe').toU8a();

const firstEncode = u8aConcat(
  operatorEncoded.toU8a(true),
  namespaceEncoded.toU8a(false),
);
const finalEncoded = u8aConcat(unsubscribeEncoded, firstEncode);

const scheduleName = blake2AsHex(finalEncoded);
```

Then you can use the calculated schedule name to call `api.query.scheduler.agenda` in order to get the block number.
Please let me know if something is unclear

Reply:
Ah now I better understand the connexion with the script, sorry for the misunderstanding @PanGan21. I'll update

Reply:
Nice, this is what i meant. I believe we can keep both examples since the example for listing scheduled unsubscribes seems also useful.